crates/xilem_core/src/sequence.rs

```diff
@@ -53,10 +53,50 @@ macro_rules! impl_view_tuple {
         }
     }
 }
-
 #[macro_export]
 macro_rules! generate_viewsequence_trait {
-    ($viewseq:ident, $view:ident, $viewmarker: ident, $bound:ident, $cx:ty, $changeflags:ty, $pod:ty; $( $ss:tt )* ) => {
+    ($viewseq:ident, $view:ident, $viewmarker: ident, $elements_splice: ident, $bound:ident, $cx:ty, $changeflags:ty, $pod:ty; $( $ss:tt )* ) => {
+
+        pub trait $elements_splice {
+            /// Push new element to the collection
+            fn push(&mut self, element: $pod);
+            /// Mutate the next existing element, and add it to this collection
+            fn mutate(&mut self) -> &mut $pod;
+            // TODO this could also track view id changes (old_id, new_id)
+            /// Mark any changes done by `mutate` on the current element
+            fn mark(&mut self, changeflags: $changeflags) -> $changeflags;
+            /// Delete the next n existing elements
+            fn delete(&mut self, n: usize);
+            /// Current length of the elements collection
+            fn len(&self) -> usize;
+            // /// skip the next n existing elements
+            // fn skip(&mut self, n: usize);
```

Comment:
Well I have put it there, because it makes sense as soon as we have reusable/immutable datastructures/ViewSequences such as ropes, where this will be needed. So I put it there as a reminder.

I don't see why this is a big no-no though (I agree though with having code comments that don't have any other sense than just being a comment/unused code, as they usually don't age well). But if it's really a deal-breaker I can remove it...

Reply:
Right now the Xilem codebase is littered with comment-out code that was supposed to be temporary reminders. Right now the consensus is slowly moving towards "if we haven't implemented it yet, don't pretend like having a commented-out stubs means it's just around the corner".

If you want to add a reminder to your code, a TODO is better, ideally with an associated issue link (for now that can be a link to this PR).

Reply:
Ok, that sounds reasonable, I've converted it into a `TODO` describing also potential use-cases for it.